### PR TITLE
Upgrade Java version for ORT Server to 21

### PR DESCRIPTION
**Pull request by oheger-bosch**
Build all artifacts with Java 21.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@
 [versions]
 # Docker
 # When updating this version make sure to keep it in sync with the worker Dockerfiles.
-eclipseTemurin = "17.0.12_7-jdk-jammy@sha256:56c8b3a8f4299e0190095333cf85e647e810bdf167e59f224a1fd432e47a8f12"
+eclipseTemurin = "21.0.4_7-jdk-jammy@sha256:0472478e22da0f66043fa6acd8cd30126592349f47937adafc2340794e5bf06a"
```
Comment (sschuberth) [inline gradle/libs.versions.toml:21]: Maybe link to https://github.com/docker-library/docs/blob/master/eclipse-temurin/README.md#supported-tags-and-respective-dockerfile-links in the commit message for reference?

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
       uses: actions/setup-java@b36c23c0d998641eff861008f374ee103c25ac73 # v4
       with:
         distribution: temurin
-        java-version: 17
+        java-version: 21
```
Comment (sschuberth) [inline .github/workflows/build-and-test.yml:27]: What about `java-version: '11'` in `.github/workflows/docker-build.yml`?

File: buildSrc/src/main/kotlin/ort-server-kotlin-jvm-conventions.gradle.kts [hunk 3]
```diff
@@ -33,16 +35,22 @@ plugins {
     id("org.jetbrains.kotlin.jvm")
 }
 
-val javaVersion = JavaVersion.current()
-val maxKotlinJvmTarget = runCatching { JvmTarget.fromTarget(javaVersion.majorVersion) }
-    .getOrDefault(enumValues<JvmTarget>().max())
+java {
+    toolchain {
```
Comment (sschuberth) [inline buildSrc/src/main/kotlin/ort-server-kotlin-jvm-conventions.gradle.kts:39]: If you take a toolchain into use, then the previous commit could have simply omitted setting up Java, as explained at https://github.com/oss-review-toolkit/ort/commit/494a32434823a3f0190db4e8e128b2efb223a14a.

Comment (sschuberth) [reply buildSrc/src/main/kotlin/ort-server-kotlin-jvm-conventions.gradle.kts:39]: And note that also the Java version used to run the Gradle build itself needs to be configured, as the build uses ORT's Detekt rules which are also built with Java 21, see https://github.com/oss-review-toolkit/ort-workbench/commit/b800aaac9fc2c36b3246f579f1ed5a48b712ef02.

File: .github/workflows/build-and-test.yml (hunk 2 above)
Comment (oheger-bosch) [reply .github/workflows/build-and-test.yml:27]: I also switched this to 21. I was not sure whether there is a specific reason why here a different Java version should be used. Maybe an oversight?

File: buildSrc/src/main/kotlin/ort-server-kotlin-jvm-conventions.gradle.kts (hunk 3 above)
Comment (oheger-bosch) [reply buildSrc/src/main/kotlin/ort-server-kotlin-jvm-conventions.gradle.kts:39]: Thanks, this should simplify the workflows.

File: .github/workflows/build-and-test.yml (hunk 2 above)
Comment (sschuberth) [reply .github/workflows/build-and-test.yml]: I guess so.

**Review by sschuberth — APPROVED** [review]

**Review by sschuberth — APPROVED** [review]

**Pull request merged**